File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        
        for(string& t : tokens){
            if(t == "+" || t == "-" || t == "*" || t == "/"){
                int b = st.top(); st.pop();
                int a = st.top(); st.pop();
                
                if(t == "+") st.push(a + b);
                else if(t == "-") st.push(a - b);
                else if(t == "*") st.push(a * b);
                else st.push(a / b);
            } else {
                st.push(stoi(t));
            }
        }
        
        return st.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>(); 
        for(String s : tokens){
            int result = 0; 
            if(s.equals("+") || s.equals("-") || s.equals("*") || s.equals("/")){
                int right = stack.pop(); 
                int left = stack.pop(); 

                if(s.equals("+")) result = left + right; 
                if(s.equals("-")) result = left - right; 
                if(s.equals("*")) result = left * right; 
                if(s.equals("/")) result = left / right; 
            } else{
                result = Integer.parseInt(s); 
            }

            stack.push(result); 
        }
        return stack.pop(); 
    }
}